app/scripts/views/sign_up.js

```diff
@@ -226,6 +231,27 @@ function (Cocktail, _, p, BaseView, FormView, Template, AuthErrors, mailcheck,
       return this._isMailcheckEnabledValue;
     },
 
+    passwordCheck: function () {
+      var self = this;
+      var password = this.getElementValue('.password');
+      try {
+        this.passwordcheck(password, function (msg) {
+          if (msg === 'MISSING_PASSWORD') {
```

Comment:
I would convert this entire block, from 239-246 to try to reduce duplication and use the functionality already provided by `AuthErrors.toError` to do the right thing:

```js
var err;
if (msg === 'MISSING_PASSWORD') {
  err = AuthErrors.toError('PASSWORD_REQUIRED');
} else {
  err = AuthErrors.toError(msg);
}
self.logError(err);
```